.github/workflows/main.yaml

```diff
@@ -46,6 +46,8 @@ jobs:
           - java_version: 23
             mvn_opts: ''
             experimental: true
+    env:
+      OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
```

Comment:
> Secrets cannot be directly referenced in if: conditionals. Instead, consider setting secrets as job-level environment variables, then referencing the environment variables to conditionally run steps in the job. 

https://docs.github.com/en/actions/security-for-github-actions/security-guides/using-secrets-in-github-actions#using-secrets-in-a-workflow